search/processor.cpp

```diff
@@ -591,21 +681,30 @@ void Processor::InitGeocoderParams(Geocoder::Params & params, bool viewportSearc
   m_geocoder.SetParams(params);
 }
 
-void Processor::InitPreRanker(bool viewportSearch)
+void Processor::InitPreRanker()
 {
   PreRanker::Params params;
 
-  params.m_accuratePivotCenter = GetPivotPoint(viewportSearch);
+  params.m_accuratePivotCenter = GetPivotPoint();
   m_preRanker.Init(params);
 }
 
-void Processor::InitRanker(bool viewportSearch)
+void Processor::InitRanker()
 {
+  size_t const kResultsCount = 30;
+  bool const viewportSearch = m_mode == Mode::Viewport;
   Ranker::Params params;
 
   params.m_currentLocaleCode = m_currentLocaleCode;
-  if (m_mode == Mode::Viewport)
+  if (viewportSearch)
+  {
     params.m_viewport = GetViewport();
+    params.m_limit = kResultsCount;
```

Comment:
This is completely rotten. See FlushViewportResults in ranker.cpp: the limit is not used in viewport search; the effective limit is 200 because that's the limit used in PreRanker.